src/components/entity/ha-battery-icon.js

```diff
@@ -0,0 +1,32 @@
+import { html } from "@polymer/polymer/lib/utils/html-tag";
+/* eslint-plugin-disable lit */
+import { PolymerElement } from "@polymer/polymer/polymer-element";
+import { batteryIcon } from "../../common/entity/battery_icon";
+import "../ha-icon";
+
+class HaBatteryIcon extends PolymerElement {
```

Comment:
I know you just copied this, but we should not add new PolymerElements, we are trying to get rid of them 😄

Reply:
What should I do instead here?

Reply:
`LitElement`

Reply:
```ts
@customElement("ha-battery-icon")
class HaBatteryIcon extends LitElement {
    
  @property() public batteryStateObj;
  @property() public batteryChargingStateObj;

  protected render() {
    return html`
      <ha-icon
        .icon=${batteryIcon(batteryStateObj, batteryChargingStateObj)}
      ></ha-icon>
    `;
  }
}
```

Reply:
Thank you. So - much - cleaner.

Reply:
Instead of using an `icon`, we should probably import the icon paths directly and use `ha-svg-icon` ?

Reply:
Yes, but that would only work for this element so would need separate logic.

Reply:
`import { mdiBatteryCharging40, mdiBatteryCharging50, .... } from '@mdi/js';`

Seems like I'd need to import about 25 items and make an object to look them up in order to implement `ha-svg-icon`, unless there is a better way??

Reply:
No, that would be the way.

The benefit of that is that we don't have to load an entire (or multiple) icon chunk of MDI icons and requests to the database where they are cached.